src/ansys/dpf/core/geometry.py

```diff
@@ -0,0 +1,403 @@
+    Parameters
+    ----------
+    center : array, list
+        3D coordinates of the center point of the plane.
+    normal : array, list, Line
+        Normal direction to the plane.
+
+    Examples
+    --------
+    Create a plane from its center and normal direction, print object information and plot.
+
+    >>> from ansys.dpf.core.geometry import Plane
+    >>> plane = Plane([0, 0, 0], [1, 0, 0])
+    >>> print(plane)
+    DPF Plane object:
+    Center point: [0, 0, 0]
+    Normal direction: [1. 0. 0.]
+    Plane has not been discretized.
+      Use plane.discretize(width, height, num_cells_x, num_cells_y)
+    >>> plane.discretize(width=1, height=1, num_cells_y=10, num_cells_x=10)
+    >>> print(plane)
+    DPF Plane object:
+    Center point: [0, 0, 0]
+    Normal direction: [1. 0. 0.]
+    Plane discretizaton using:
+      Width (x-dir): 1
+      Height (y-dir): 1
+      Num cells x-dir: 10
+      Num cells y-dir: 10
+    >>> plane.plot()
+
+    """
+
+    def __init__(self, center, normal, server=None):
+        """Initialize Plane object from its center and normal direction."""
+        # Input check
+        if not len(center) == 3:
+            raise ValueError("'center' of the plane must have length 3")
+        if not isinstance(normal, Line):
+            if len(normal) == 2:
+                if not len(normal[0]) == len(normal[1]) == 3:
+                    raise ValueError("Each point must contain 3 coordinates.")
+                normal_vect = normal
+                normal_dir = self._get_direction_from_vect(normal_vect)
+            elif len(normal) == 3:
+                normal_dir = normal / np.linalg.norm(normal)
+                normal_vect = [np.array([0, 0, 0]), normal_dir]
+        else:
+            normal_vect = [normal.coordinates.data[0], normal.coordinates.data[1]]
+            normal_dir = self._get_direction_from_vect(normal_vect)
+
+        self._center = center
+        self._normal_vect = normal_vect
+        self._normal_dir = normal_dir
+        self._server = server
+        self._mesh = None
+        self.width = None
+        self.height = None
+        self.num_cells_x = None
+        self.num_cells_y = None
+
+    def __str__(self):
+        """Print plane information."""
+        txt = "DPF Plane object:\n"
+        txt += f"Center point: {self._center}\n"
+        txt += f"Normal direction: {self._normal_dir}\n"
+        if self._mesh:
+            txt += "Plane discretizaton using:\n"
+            txt += f"  Width (x-dir): {self.width}\n"
+            txt += f"  Height (y-dir): {self.height}\n"
+            txt += f"  Num cells x-dir: {self.num_cells_x}\n"
+            txt += f"  Num cells y-dir: {self.num_cells_y}\n"
+        else:
+            txt += "Plane has not been discretized.\n"
+            txt += "  Use plane.discretize(width, height, num_cells_x, num_cells_y)\n"
+        return txt
+
+    @property
+    def center(self):
+        """Center of the plane."""
+        return self._center
```

Comment:
A plane has center? I thought they were infinite.

Reply:
Yes, but I decided to define a plane given its center and normal. Hence, we need to store and retriever these two variables from the object.

Reply:
A very valid reason!